pkg/api/apiloader.go

```diff
@@ -4,6 +4,7 @@ import (
 	"encoding/json"
 	"io/ioutil"
 
+	"github.com/Azure/acs-engine/pkg/agentPoolOnlyApi/v20170831"
```

Comment:
It still makes sense for all "agentPoolOnlyApi" to be under separate folder (could be moved under api) but shouldn't be a peer of container service type. 

The reason is both api models support similarly named fields/properties but shouldn't necessarily be shared because each of those properties could have their own fields and could cause conflicts it cs and cp types are under the same version (e.g. ServicePrincipalProfile). Placing them directly under "github.com/Azure/acs-engine/pkg/api/v20170831" means cs and cp resource types could be forced to be in the same package name and that's not desired.